superset/charts/api.py

```diff
@@ -283,7 +283,7 @@ def post(self) -> Response:
             return self.response_400(message=error.messages)
         try:
             new_model = CreateChartCommand(g.user, item).run()
-            return self.response(201, id=new_model.id, result=item)
+            return self.response(201, id=new_model.id, result=new_model.to_json())
```

Comment:
Since the model is mutated on the command, we need to return the final metadata, not the original one.